### PR TITLE
Added validation to allow only corporate emails to book

### PR DESCRIPTION
**Pull request by steverenardbroadsign**

### CODE REVIEW

File: Book-A-Desk.Domain/Reservation.Validator.fs [hunk 1]
```diff
@@ -2,19 +2,23 @@
...
-        match email with
-        | EmailAddress "" ->
-            Error "The e-mail address must not be empty."
-        | _ ->
-            Ok ()
+    let private validateCorporateEmail email =
+        let emailToValidate = email.ToString().ToLower()
+        let emailValidator = EmailAddressAttribute()
+        let isValidEmail = emailValidator.IsValid(emailToValidate)
+        let hasCorporateDomain = Regex.Match(emailToValidate, "@broadsign.com")
```
Comment (jimmy-cyr) [inline Book-A-Desk.Domain/Reservation.Validator.fs]: this project is open source so we should probably inject the corporate domain

File: Book-A-Desk.Domain/Reservation.Validator.fs [hunk 2]
```diff
@@ -51,7 +55,7 @@ module BookADeskReservationValidator =
     }
             
     let validateCommand (offices: Office list) (cmd : BookADesk) reservationAggregate = result {
-        do! validateEmailIsNotEmpty cmd.EmailAddress
```
Comment (serherca) [inline Book-A-Desk.Domain/Reservation.Validator.fs:65]: Should we keep validating that email is not empty? I don't know if your validation shows an error when email is empty.

File: Book-A-Desk.Domain/Reservation.Validator.fs (hunk 1 above)
Comment (AlexGagne) [reply Book-A-Desk.Domain/Reservation.Validator.fs]: +1

File: Book-A-Desk.Domain/Reservation.Validator.fs [hunk 3]
```diff
@@ -2,19 +2,23 @@
...
-            Ok ()
+    let private validateCorporateEmail email =
+        let emailToValidate = email.ToString().ToLower()
+        let emailValidator = EmailAddressAttribute()
+        let isValidEmail = emailValidator.IsValid(emailToValidate)
+        let hasCorporateDomain = Regex.Match(emailToValidate, "@broadsign.com")
+        if isValidEmail && hasCorporateDomain.Success then
+            Ok()
+        else
+            Error "The e-mail address must be valid."
```
Comment (AlexGagne) [inline Book-A-Desk.Domain/Reservation.Validator.fs]: "The e-mail address is invalid." ?

**Review by AlexGagne — APPROVED** [review]
Good, but like Jimmy said, we should probably inject the corporate domain. Either make it quite obvious for any fork to modify or pass it as an environment variable.

File: Book-A-Desk.Domain/Reservation.Validator.fs (hunk 1 above)
Comment (steverenardbroadsign) [reply Book-A-Desk.Domain/Reservation.Validator.fs:17]: Good point. Feel free to update the PR

File: Book-A-Desk.Domain/Reservation.Validator.fs (hunk 2 above)
Comment (steverenardbroadsign) [reply Book-A-Desk.Domain/Reservation.Validator.fs:65]: This implementation validates empty email (you will see that in the test cases of InvalidEmails).

File: Book-A-Desk.Domain/Reservation.Validator.fs (hunk 1 above)
Comment (steverenardbroadsign) [reply Book-A-Desk.Domain/Reservation.Validator.fs]: @jimmy-cyr I will take care, good point.

File: Book-A-Desk.Domain/Reservation.Validator.fs (hunk 3 above)
Comment (steverenardbroadsign) [reply Book-A-Desk.Domain/Reservation.Validator.fs]: Updated

File: Book-A-Desk.Domain/Reservation.Validator.fs (hunk 1 above)
Comment (steverenardbroadsign) [reply Book-A-Desk.Domain/Reservation.Validator.fs]: I just pushed the commit to implement that.

File: Book-A-desk.Domain.Tests/Reservation.Validator.Tests.fs [hunk 4]
```diff
@@ -79,12 +113,12 @@ let ``GIVEN A Book-A-Desk Reservation command WITH a date greater than today, WH
 let ``GIVEN A Book-A-Desk Reservation command WITH an invalid office id, WHEN validating, THEN validation should fail`` () =
     let commandWithInvalidOfficeId =
         {
-            EmailAddress = EmailAddress "anEmailAddress@fake.com"
+            EmailAddress = EmailAddress $"email@{domainName}"
```
Comment (AlexGagne) [inline Book-A-desk.Domain.Tests/Reservation.Validator.Tests.fs:116]: Not for this PR:
An idea: the validation for the email address could be moved in the EmailAddress constructor so it's impossible to have an invalid email in the domain

File: Book-A-Desk.Api/Program.fs [hunk 5]
```diff
@@ -37,11 +37,12 @@ let configureApp (ctx : WebHostBuilderContext) (app : IApplicationBuilder) =
 
     let getAllOffices = (fun () -> Offices.All)
 
-    let reservationCommandsFactory = ReservationCommandsFactory.provide getAllOffices
+    let validDomainName = ctx.Configuration.["DomainName"]
```
Comment (AlexGagne) [inline Book-A-Desk.Api/Program.fs:40]: Make sure this is in the parameter store.

**Pull request merged**